src/app/student/map/page.tsx

```diff
@@ -0,0 +1,51 @@
+import MainView from "@/app/student/map/_components/MainView"
+import {
+  BoothMap,
+  geoJsonBoothData,
+  makeBooth
+} from "@/app/student/map/lib/booths"
+import { LocationId, locations } from "@/app/student/map/lib/locations"
+import { feature } from "@/components/shared/feature"
+import { fetchExhibitors } from "@/components/shared/hooks/api/useExhibitors"
+import { notFound } from "next/navigation"
+import { Suspense } from "react"
+
+export default async function Page() {
+  if (!feature("MAP_PAGE")) {
+    return notFound()
+  }
+
+  const exhibitors = await fetchExhibitors({
+    year: 2024,
+    next: { revalidate: 3600 * 24 * 6 /* 6 days */ }
+  })
+
+  const exhibitorsByID = new Map(exhibitors.map(e => [e.id, e]))
+
+  const boothsById: BoothMap = new Map(
+    geoJsonBoothData.features.map(feat => [
+      feat.properties.id,
+      makeBooth(feat, exhibitorsByID)
+    ])
+  )
+
+  const boothsByLocation: Map<LocationId, BoothMap> = new Map(
+    locations.map(loc => [loc.id, new Map()])
+  )
+  boothsById.forEach((booth, id) => {
+    boothsByLocation.get(booth.location)!.set(id, booth)
+  })
+
+  return (
+    // TODO: pt-16 is to account for the navbar, will break if navbar size changes
+    <Suspense>
```

Comment:
Since the entire page is in the suspense does it provide any value to have it? Maybe add a fallback, or extract parts of the page outside the suspense

Reply:
Yes the previous Vercel deployment error showed up when we dont have it, that's why I added it here. It caused by calling useSearchParams() function in MainView.tsx, so basically the whole page is suspense, I could not come up a best extraction for only certain part of the page by now.